api/v1beta1/metal3data_types.go

```diff
@@ -35,7 +35,8 @@ type Metal3DataSpec struct {
 
 	// TemplateReference refers to the Template the Metal3MachineTemplate refers to.
 	// It can be matched against the key or it may also point to the name of the template
-	// Metal3Data refers to
+	// Metal3Data refers to.
+	// Deprecated: This field is deprecated and will be removed in a future release.
```

Comment:
I think we need an empty doc line to make the deprecation notice be picked up by tooling (linters, controller-runtime) as seen in [this CAPI PR](https://github.com/kubernetes-sigs/cluster-api/pull/11741/files)
```suggestion
        //
	// Deprecated: This field is deprecated and will be removed in a future release.
```